### DISCUSSION TITLE
Publishing from RaspberryPi successful but not seeing the messages under topic in AWS IoT Hub

### DISCUSSION THREAD

**Discussion by AnushaNagesh97** [aws/aws-iot-device-sdk-python#342] [Q&A]
Hi! 

I am using AWS IoT hub for the first time. After setting up the Thing and policy, and downloading the connection kit, I was able to see messages published to the sdk/test/python topic. However I am unable to see messages published to any other topics from my RPi5. I have provided the necessary certificate files, private key and endpoint. I have tried with TLS versions 1.2 and 23. All my error catch messages show that no errors were caught on my MQTT client end with respect to connection, publishing or certificate location. 

Could you provide insight on the potential issues that could be causing this?

```python
import time
import paho.mqtt.client as mqtt
import ssl
import os

def on_connect(client, userdata, flags, rc):
    if rc == 0:
        print("Connected to AWS IoT Core successfully.")
        client.subscribe("camdata")
        publish_message(client)
    else:
        print(f"Connection error: {rc}")
        # Handle connection error (e.g., retry connection)

def publish_message(client):
    try:
        client.publish("camdata", payload="hi", qos=1, retain=True)  # Changed QoS to 1
        print("Published 'hi' to AWS IoT Core")
    except Exception as e:
        print(f"Error publishing message: {e}")
        # Handle publishing error (e.g., log the error)

def main():
    try:
        # Print the current working directory
        current_directory = os.getcwd()
        print("Current working directory:", current_directory)

        # Replace with your AWS IoT Core credentials
        aws_iot_core_endpoint = "//redacted//"
        ca_certs = "./root-CA.crt"
        certfile = "./rasppi5.cert.pem"
        keyfile = "./rasppi5.private.key"

        if not all(os.path.exists(f) for f in [ca_certs, certfile, keyfile]):
            print("Error: Certificate file(s) not found.")
            return

        client = mqtt.Client()
        client.on_connect = on_connect

        # Configure TLS with valid certificates
        try:
            client.tls_set(ca_certs=ca_certs, certfile=certfile, keyfile=keyfile, tls_version=ssl.PROTOCOL_TLSv1_2)
        except FileNotFoundError as e:
            print(f"Error: Certificate file not found - {e}")
            return  # Exit the program if certificates are missing

        # Connect to AWS IoT Core
        client.connect(aws_iot_core_endpoint, 8883, 60)

        # Start the network loop (this will run in a background thread)
        client.loop_start()

        # Publish a single message and wait
        time.sleep(5)  # Add delay to ensure connection and subscription
        publish_message(client)
        print("Message published. Waiting for acknowledgment.")

        # Gracefully stop the network loop and disconnect
        client.loop_stop()
        client.disconnect()

    except Exception as e:
        print(f"Unhandled error: {e}")

if __name__ == "__main__":
    main()
```

**Comment by jmklix** [comment]
You're only subscribing to one topic in the above code:
```python
client.subscribe("camdata")
```
You will only ever see messages published to "camdata" while running this sample. You can subscribe to multiple things and/or use wildcards to subscribe to many topics at once. You can read more about MQTT topics [here](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html).

Please let me know if you have any other questions about how to use this sdk. We do recommend using the [v2 sdk](https://github.com/aws/aws-iot-device-sdk-python-v2) as it is newer and better.

**Reply by AnushaNagesh97** [reply]
I am not receiving any messages under camdata, or #. What could be the issue here?

**Reply by jmklix** [reply]
@github-actions proposed-answer

It's also possible that your the policy you have attached to your IoT thing is limited in what it can publish to. You can set it to allow all publishes and subscriptions while testing with this:
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "*"
      ],
      "Resource": [
        "*"
      ]
    }
  ]
}
```
You can read more about IoT policies [here](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html)
